src/schemas.jl

```diff
@@ -250,6 +253,9 @@ otherwise constitutes type piracy and should be avoided.
 @inline accepted_field_type(::SchemaVersion, T::Type) = T
 accepted_field_type(::SchemaVersion, ::Type{UUID}) = Union{UUID,UInt128}
 accepted_field_type(::SchemaVersion, ::Type{Symbol}) = Union{Symbol,String}
+accepted_field_type(::SchemaVersion, ::Type{Vector{String}}) = AbstractVector{<:AbstractString}
+accepted_field_type(::SchemaVersion, ::Type{Vector{T}}) where T = AbstractVector{<:T}
```

Comment:
```suggestion
accepted_field_type(sv::SchemaVersion, ::Type{<:Vector{T}}) where T = AbstractVector{<:(accepted_field_type(sv, T))}
```

?

i think this should also obviate the need for the other `accepted_field_type` definitions for `Vector` here

Reply:
When `accepted_field_type(::SchemaVersion, ::Type{Vector}) = AbstractVector` is removed, we still hit error in [this test](https://github.com/beacon-biosignals/Legolas.jl/blob/321b19c88c5c7e07c616278f2aaf876495533667/test/runtests.jl#L468):
```julia
┌ Warning: The `Tables.Schema` of the `Arrow.Table` read via `Legolas.read(io_or_path)` does not appear to
│ comply with the `Legolas.SchemaVersion` indicated by the table's metadata (`SchemaVersion("test.parent", 1)`). Try invoking
│ `Legolas.read(io_or_path; validate=false)` to inspect the table.
└ @ Legolas ~/Code/Legolas.jl/src/tables.jl:171
miscellaneous Legolas/src/tables.jl tests: Error During Test at /Users/hrobertson/Code/Legolas.jl/test/runtests.jl:468
  Test threw exception
  Expression: t == [NamedTuple(ParentV1(r)) for r = Tables.rows(Legolas.read(path))]
  ArgumentError: Tables.Schema violates Legolas schema `test.parent@1`:
   - Incorrect type: `x` expected `<:Vector`, found `SubArray{Int64, 1, Arrow.Primitive{Int64, Vector{Int64}}, Tuple{UnitRange{Int64}}, true}`
  Provided Tables.Schema:
   :x  SubArray{Int64, 1, Arrow.Primitive{Int64, Vector{Int64}}, Tuple{UnitRange{Int64}}, true}
   :y  String
  Stacktrace:
   [1] validate(ts::Tables.Schema{(:x, :y), Tuple{SubArray{Int64, 1, Arrow.Primitive{Int64, Vector{Int64}}, Tuple{UnitRange{Int64}}, true}, String}}, sv::SchemaVersion{Symbol("test.parent"), 1})
```

Reply:
yeah, have to keep that one actually, my bad